src/D2L.Bmx/Aws/AwsClient.cs

```diff
@@ -0,0 +1,75 @@
+using System.Text;
+using System.Web;
+using System.Xml;
+using Amazon.SecurityToken;
+using Amazon.SecurityToken.Model;
+
+namespace D2L.Bmx.Aws;
+
+public interface ICloudProvider<TRoleState> where TRoleState : IRoleState {
```

Comment:
* the heavy use of generics and abstraction is one aspect of the `dotnet` branch that we want to get away from by this new rewrite
* we should default to making types internal rather than public

```suggestion
internal interface IAwsClient {
```